go/vt/mysqlctl/mysqld.go

```diff
@@ -107,9 +114,53 @@ func NewMysqld(dbcfgs *dbconfigs.DBConfigs) *Mysqld {
 	result.appPool = dbconnpool.NewConnectionPool("AppConnPool", *appPoolSize, *appIdleTimeout, *poolDynamicHostnameResolution)
 	result.appPool.Open(dbcfgs.AppWithDB(), appMysqlStats)
 
+	version, _ := result.getVersionString()
+	result.flavor, result.version = result.DetectFlavorVersion(version)
+	result.c = NewCapabilitySet(result)
+
+	log.Infof("mysqld is flavor: %s, version: %v", result.flavor, result.version)
 	return result
 }
 
+func (mysqld *Mysqld) getVersionString() (string, error) {
+	mysqlRoot, err := vtenv.VtMysqlRoot()
+	if err != nil {
+		return "", err
+	}
+	mysqldPath, err := binaryPath(mysqlRoot, "mysqld")
+	if err != nil {
+		return "", err
+	}
+	_, version, err := execCmd(mysqldPath, []string{"--version"}, nil, mysqlRoot, nil)
+	if err != nil {
+		return "", err
+	}
+	return version, nil
+}
+
+// DetectFlavorVersion takes the output of mysqld --version
+// and parses it into a flavor and version
+func (mysqld *Mysqld) DetectFlavorVersion(version string) (flavor mysqlFlavor, ver serverVersion) {
+	if strings.Contains(version, "MySQL") {
+		flavor = flavorMySQL
+	} else if strings.Contains(version, "Percona") {
+		flavor = flavorPercona
+	} else if strings.Contains(version, "MariaDB") {
+		flavor = flavorMariaDB
+	} else {
+		log.Fatalf("unrecognized server flavor from: %s", version)
+	}
+	v := versionRegex.FindStringSubmatch(version)
+	if len(v) != 4 {
+		log.Fatalf("Could not parse server version from: %s", version)
+	}
+	ver.Major, _ = strconv.Atoi(string(v[1]))
```

Comment:
Didn't see a change here. Did you resolve by mistake?

Reply:
No, it was changed to:
```
		return flavor, ver, fmt.Errorf("Could not parse server version from: %s", version)
```

Reply:
Ah sorry I should have specified. I meant also check each error from the 3 calls to `Atoi()`. Granted, those shouldn't fail because we matched the regexp, but I prefer to check all errors. Someone might come along and change things later so that those Atoi() calls might be capable of failing.